File: Python/Easy/strings/maxScoreAfterSplittingString.py
def solution(s):
    max_score = 0
    ones_right = s.count('1')
    zeros_left = 0

    for i in range(len(s) -1):
        zeros_left += s[i] == '0'
        ones_right -= s[i] == '1'
        max_score = max(max_score, zeros_left+ones_right)

    return max_score

File: JavaScript/Easy/strings/maxScoreAfterSplittingString.js
var solution = function(s){
    let maxScore = 0
    let zerosLeft = 0;
    let onesRight = s.match(/1/g)
    if (onesRight === null){
        return s.length -1
    } else {
        onesRight = onesRight.length
    }

    for (let i = 0; i< s.length-1; i++){
        s[i] === '1' ? onesRight -- : onesRight
        s[i] === '0' ? zerosLeft ++ : zerosLeft
        maxScore = Math.max(maxScore, zerosLeft + onesRight)
    }

    return maxScore
}